openslides_backend/action/actions/user/create_update_permissions_mixin.py

```diff
@@ -1,150 +1,424 @@
         ],
+        "B": [
+            "number_$",
+            "structure_level_$",
+            "vote_weight_$",
+            "about_me_$",
+            "comment_$",
+            "vote_delegated_$_to_id",
+            "vote_delegations_$_from_ids",
+        ],
+        "C": ["group_$_ids"],
+        "D": ["committee_ids", "committee_$_management_level"],
+        "E": ["organisation_management_level"],
+        "F": ["is_demo_user"],
     }
 
     def check_permissions(self, instance: Dict[str, Any]) -> None:
         """
-        Precondition:
-         If there is more than one right sufficient for one field (Group C), the Meeting-Level-Permissions
-         are expected to be the last in the field_rights list
+        Checks the permissions on a per field and user.scope base, details see
+        https://github.com/OpenSlides/OpenSlides/wiki/user.update or user.create
+        The fields groups and their necessary permissions are also documented there.
         """
         self.assert_not_anonymous()
 
-        user = self.datastore.get(
-            FullQualifiedId(Collection("user"), self.user_id),
-            ["organisation_management_level", "committee_as_manager_ids"],
-        )
-        user_oml = OrganisationManagementLevel(
-            user.get("organisation_management_level")
-        )
-        if user_oml == OrganisationManagementLevel.SUPERADMIN:
+        permstore = PermissionVarStore(self.datastore, self.user_id)
+        if permstore.user_oml == OrganisationManagementLevel.SUPERADMIN:
+            return
+
+        self._check_OML_in_instance(permstore, instance)
+        actual_group_fields = self._get_actual_grouping_from_instance(instance)
+
+        self.check_group_E(permstore, actual_group_fields["E"])
+        self.check_group_D(permstore, actual_group_fields["D"], instance)
+        self.check_group_C(permstore, actual_group_fields["C"], instance)
+        self.check_group_B(permstore, actual_group_fields["B"], instance)
+        self.check_group_A(permstore, actual_group_fields["A"], instance)
+        self.check_group_F(actual_group_fields["F"])
+
+    def check_group_A(
+        self, permstore: PermissionVarStore, fields: List[str], instance: Dict[str, Any]
+    ) -> None:
+        """ Check Group A common fields: Depending on scope of user to act on """
+        if (
+            not fields
+            or permstore.user_oml >= OrganisationManagementLevel.CAN_MANAGE_USERS
+        ):
             return
 
+        if uid := instance.get("id"):
+            self.datastore.update_additional_models(
+                FullQualifiedId(Collection("user"), uid), instance
+            )
+
+        scope, scope_id = self.get_user_scope(uid, None if uid else instance)
+        if scope == UserScope.Organisation:
+            raise MissingPermission({OrganisationManagementLevel.CAN_MANAGE_USERS: 1})
+        elif scope == UserScope.Committee:
+            if scope_id not in permstore.user_committees(
```

Comment:
This is also affected by the comment above: As far as I know does the CML can be set, but the user may not be part of the committee anymore.

Reply:
user_committee is now a property with all committees of the request user, where the user has manage permission, which now includes beeing part of the committee.